### PR TITLE
Wrap-up Modal and Sideframe

### PR DESCRIPTION
**Pull request by FinalAngel**
this pr addresses the following issues:

- cleanup modal code
- add close functionality to dimmer

note: somehow @vxsx sneaked in here

### CODE REVIEW

File: cms/static/cms/js/modules/cms.modal.js [hunk 1]
```diff
@@ -439,34 +420,32 @@ var CMS = window.CMS || {};
...
-                // cancel action when minimized
-                if (this.minimized) {
+                // cancel if maximized or minimized
+                if (this.maximized || this.minimized) {
                     return false;
                 }
 
                 var that = this;
                 var position = that.ui.modal.position();
+                var left;
```
Comment (vxsx) [inline cms/static/cms/js/modules/cms.modal.js]: ooh, smart

File: cms/static/cms/js/modules/cms.modal.js [hunk 2]
```diff
@@ -578,7 +556,10 @@ var CMS = window.CMS || {};
                     // check if the item is the last one
                     var last = (index >= breadcrumbs.length - 1) ? 'active' : '';
                     // render breadcrumbs
-                    crumb += '<a href="' + item.url + '" class="' + last + '"><span>' + item.title + '</span></a>';
+                    crumb += template
+                        .replace('{1}', item.url)
+                        .replace('{2}', last)
```
Comment (vxsx) [inline cms/static/cms/js/modules/cms.modal.js]: potentially after this replace there could be multiple `{3}` in the string
kinda unlikely, but yeah

File: cms/static/cms/js/modules/cms.modal.js [hunk 3]
```diff
@@ -645,12 +624,13 @@ var CMS = window.CMS || {};
                         cls = 'cms-btn cms-btn-caution';
                     }
 
-                    // create the element and attach events
                     var el = $('<a href="#" class="' + cls + ' ' + item.attr('class') + '">' + title + '</a>');
+
                     el.on(that.click, function () {
                         if (item.is('input') || item.is('button')) {
                             item[0].click();
```
Comment (vxsx) [inline cms/static/cms/js/modules/cms.modal.js]: `trigger('click')` ?

File: cms/static/cms/js/modules/cms.modal.js (hunk 2 above)
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.modal.js]: ideally adding some template snippet (like riesig) will make sense at some point. Still better than before

File: cms/static/cms/js/modules/cms.sideframe.js [hunk 4]
```diff
@@ -109,6 +114,11 @@ var CMS = window.CMS || {};
              * @param [animate] {Number} Animation speed in ms
              */
             open: function open(url, animate) {
+                if (!url) {
+                    throw new Error('The arguments passed to "open" were invalid.');
```
Comment (vxsx) [inline cms/static/cms/js/modules/cms.sideframe.js]: <3 but we should at some point only do this for --debug environments

**Comment by vxsx** [comment]
LGTM

File: cms/static/cms/js/modules/cms.sideframe.js (hunk 4 above)
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.sideframe.js]: yep

**Comment by vxsx** [comment]
LGTM

**Pull request merged**